Suppress back-up-seed prompt when already done (fixes #8938)

acolytec3 opened the pull request:
fixes #8938 

### Summary

Suppresses backup-seed-phrase option on Pirvacy & Security screen when already backed up

#### Platforms
- Android
- iOS

#### Areas that maybe impacted
Privacy & Security Settings

##### Functional

- Privacy & Security Settings

### Steps to test
1. Open Status
2.  Open Profile
3. Open Privacy & Security Settings
4. Back up seed phrase 
5. Navigate back to Privacy & Security Settings
6.  Security section will be suppressed

status: ready 

Visual representation on Android
![Back-up-seed-suppresed](https://user-images.githubusercontent.com/17355484/66163449-5dd1df80-e5fe-11e9-964c-e607ea2573a5.png)

acolytec3 committed:
Suppress back-up-seed prompt when already done

Signed-off-by: Andrey Shovkoplyas <motor4ik@gmail.com>

acolytec3 commented:
To reviewers, is the preference to suppress the whole security section when the backup seed has been saved, just the backup-seed menu option, or else to present the backup-seed option disabled?

As currently coded, it suppresses the whole section since this is what the original issue states as the preferred outcome.

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 40e47221 | [#1](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-9105/1/) | 2019-10-03 21:04:42 | ~10 min | `ios` | [:package:`ipa`](https://i.diawi.com/ScnRvG) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A0&chl=https%3A%2F%2Fi.diawi.com%2FScnRvG)|

flexsurfer commented on src/status_im/ui/screens/privacy_and_security_settings/views.cljs:
```diff
@@ -12,20 +12,21 @@
+      :on-press
+      #(re-frame/dispatch [:navigate-to :backup-seed])
+      :accessories
+      [(when show-backup-seed? [components.common/counter {:size 22} 1])
```
we don't need this show-backup-seed? condition then

acolytec3 replied:
@flexsurfer  Removed the condition.  Secondary question not related to the main PR, what's the point of the counter?  It doesn't appear to be rendering in the UI.

flexsurfer replied:
@andmironov ?

acolytec3 replied:
Nevermind, I figured it out.  It only shows when you haven't actually backed up your seedphrase (as in the design below from @andmironov).  Just me being clueless.

acolytec3 commented:
@flexsurfer I fixed the merge conflict and it looks another security option around biometric auth got added so I moved the parens to just be around the backup seed phrase option.  Unfortunately,my local environment has gotten hosed somehow so I'm not able to run the app locally so can't produce a new visual.  Once I get it rebuilt, I'll post one but it may be a while.

andmironov commented:
Hey! Thanks for bringing this up!

What if we do something like this? But I have not considered all the possible security implications of this? Is this some sort of vulnerability, to show that the seed phrase has been already backed up? 


<img width="834" alt="Screen Shot 2019-10-04 at 14 01 12" src="https://user-images.githubusercontent.com/3115606/66206138-7ea63d80-e6af-11e9-83ef-8d6e46c3a666.png">

acolytec3 commented:
@andmironov That should be pretty straight-forward.  Let me get my dev environment rebuilt and I can code that.  @flexsurfer What do you think of that approach?

andmironov commented:
UPD let's just grey it out with no "Done" not to confuse anyone

<img width="825" alt="Screen Shot 2019-10-04 at 15 08 31" src="https://user-images.githubusercontent.com/3115606/66209963-ee6cf600-e6b8-11e9-9cb0-08486d9b3ce6.png">

~figma https://www.figma.com/file/bPS9GrvMr7LnH7vnkmuveQfd/Settings?node-id=576%3A0

acolytec3 commented:
@flexsurfer @andmironov Is there an obvious way to override the default styling on the :title element in in the list-items section?  I tried applying a conditional {:style {:color colors/gray}} to it and it threw an error.  I guess I can always just show an alternate view using a nested-text element in that row but was curious if I'm missing something obvious.

flexsurfer commented:
@acolytec3 you can use `:disabled?` prop

churik commented:
@acolytec3 
all builds are failed.
Can you take a look?

flexsurfer commented on src/status_im/ui/screens/privacy_and_security_settings/views.cljs:
```diff
@@ -16,18 +16,18 @@
+   {:type                    :small
     :title                   :t/back-up-seed-phrase
     :accessibility-label     :back-up-recovery-phrase-button
+    :disabled?               (if show-backup-seed? false true)
```
you can use `(not show-backup-seed?)`

acolytec3 replied:
Done.  *newb silliness*

acolytec3 commented:
@andmironov @flexsurfer Here's a screengrab from the Android Emulator with the disabled prop and the chevron suppressed.  Does this look about right?

![Disabled seed phrase](https://user-images.githubusercontent.com/17355484/66226317-7eef0a80-e6a8-11e9-9112-41ab63470024.png)

@churik I'm not sure what was happening before but it seems like the builds are working again with my latest push?  I don't have access to the jenkins logs so can't tell you specifically what's wrong if anything.

flexsurfer commented on src/status_im/ui/screens/privacy_and_security_settings/views.cljs:
```diff
@@ -19,14 +19,15 @@
     :accessories
-    [(when show-backup-seed? [components.common/counter {:size 22} 1])
-     :chevron]}
+    [(when show-backup-seed? [components.common/counter {:size 22} 1]
```
this won't work, you need to move `when` outside vector `:accessories (when show-backup-seed? [[components.common/counter {:size 22} 1] :chevron])`

acolytec3 replied:
Moved in latest commit.  Will squash and force push.

churik commented:
Tested (IOS 11.4.1, Android 8):
- interruption  of the process of "Back up seed phrase" on different stages;
- creating account - back up seed phrase
- restoring account
- creating account with keycard
- restoring account with keycard

Failed e2e is not related to this PR.

Thank you for excellent job @acolytec3 !
